app/lib/dfe_data_tables/categories_parser.rb

```diff
@@ -42,17 +42,21 @@ def categories
 
         concept_hash = concept(row)
 
-        if row[3].present?
+        l0 = obj&.last
+        l1 = l0&.dig(:subcat)&.last
+        l2 = l1&.dig(:subcat)&.last
+        l3 = l2&.dig(:subcat)&.last
+        unless l3.nil?
           obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(concept_hash)
           next
         end
 
-        if row[2].present?
+        unless l2.nil?
           obj&.last&.dig(:subcat)&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(concept_hash)
           next
         end
 
-        if row[1].present?
+        unless l1.nil?
           obj&.last&.dig(:subcat)&.last&.dig(:concepts)&.push(concept_hash)
           next
         end
```

Comment:
This looks like it (and the blocks above it) add the concept to each category/subcategory. The concept should only be added to the deepest subcategory, so updating this to `if/else if` blocks might resolve the issue of concepts appearing at multiple levels of the tree.

Reply:
You made me doubt, but it's actually working as intended and fixes the bug.
1. the condition goes from checking if there is a category at that level in the incoming row, to checking if there is a category at that level already saved in the aggregator object (see `Demographics - SC` sheet in the `Information Architecture - Treejack SC, PL Recovered v2.xlsx` file, line 7, for an example in which the previous condition was causing issues).
2. the checks are performed outermost to innermost, so if a level 3 category is found, that will be the first match
3. the `next` inside the match makes sure that if a concept is assigned to a category for a level, it's not assigned to the parent categories